pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,527 @@
+                'CIM_ObjectManager'
+            new_instance['SystemName'] = 'Mock_Test_WBEMServerTest'
+            new_instance['SystemCreationClassName'] = 'CIM_ComputerSystem'
+
+        # Test all key properties in instance. The CIM repository
+        # cannot add values for key properties and does
+        # not allow creating key properties from class defaults.
+        key_props = [p.name for p in six.itervalues(target_class.properties)
+                     if 'key' in p.qualifiers]
+        for pn in key_props:
+            if pn not in new_instance:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Key property {0!A} not in NewInstance ", pn))
+
+        # Exception if property in instance but not class or types do not
+        # match
+        for iprop_name in new_instance:
+            if iprop_name not in target_class.properties:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Property {0!A} specified in NewInstance is not "
+                            "exposed by class {1!A} in namespace {2!A}",
+                            iprop_name, target_class.classname, namespace))
+
+            cprop = target_class.properties[iprop_name]
+            iprop = new_instance.properties[iprop_name]
+            if iprop.is_array != cprop.is_array or \
+                    iprop.type != cprop.type:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Instance and class property {0!A} types "
+                            "do not match: instance={1!A}, class={2!A}",
+                            iprop_name, iprop, cprop))
+
+            # The class and instnames are the same except for possible case
+            # sensitivity. If case different, set cprop_name into new instance
+            # to maintain case equality
+            cprop_name = cprop.name
+            if cprop_name != iprop_name:
+                new_instance.properties[iprop_name].name = cprop_name
+
+        # If property not in instance, add it from class and use default value
+        # from class
+        for cprop_name in target_class.properties:
+            if cprop_name not in new_instance:
+                default_value = target_class.properties[cprop_name]
+                new_instance[cprop_name] = default_value
+
+        # Build instance path. We build the complete instance path
+        new_instance.path = CIMInstanceName.from_instance(
+            target_class,
+            new_instance,
+            namespace=namespace)
+
+        # Check for duplicate instances
+        if instance_store.exists(new_instance.path):
+            raise CIMError(
+                CIM_ERR_ALREADY_EXISTS,
+                _format("NewInstance {0!A} already exists in namespace "
+                        "{1!A}.", new_instance.path, namespace))
```

Comment:
This check is already done in `instance_store.create()`, raising `ValueError`, so instead of doing this check it could be an exception handling around `instance_store.create()` that transforms the ValueError into a CIMError.

Reply:
Good catch.  Modified. DONE